content/docs/components/misc/metadata.md

```diff
@@ -4,43 +4,172 @@ description = "Tracking and managing metadata of machine learning workflows in K
+artifact in the **Artifact Store** on the Kubeflow UI.
+
+1. Go to Kubeflow in your browser. (If you haven't yet opened the 
+  Kubeflow UI, find out how to [access the
+  Kubeflow UIs](https://www.kubeflow.org/docs/other-guides/accessing-uis/).)
+1. Click **Artifact Store** in the left-hand navigation panel:
+  <img src="/docs/images/metadata-ui-option.png" 
+    alt="Metadata UI"
+    class="mt-3 mb-3 border border-info rounded">
+
+1. The **Artifacts** screen opens and displays a list of items for all the
+  metadata events that your workflows have logged. You can click the name of 
+  each item to view the details. 
+  
+    The following examples show the items that appear when you run the 
+    `demo.ipynb` notebook described [above](#demo-notebook):
+
+    <img src="/docs/images/metadata-artifacts-list.png" 
+    alt="A list of metadata items"
+    class="mt-3 mb-3 border border-info rounded">
+
+  * Example of **model** metadata with the name "MNIST":
+
+        <img src="/docs/images/metadata-model.png" 
+          alt="Model metadata for an example MNIST model"
+          class="mt-3 mb-3 border border-info rounded">
+
+  * Example of **metrics** metadata with the name "MNIST-evaluation":
+
+        <img src="/docs/images/metadata-metrics.png" 
+          alt="Metrics metadata for an evaluation of an MNIST model"
+          class="mt-3 mb-3 border border-info rounded">
+
+  * Example of **dataset** metadata with the name "mytable-dump":
+
+        <img src="/docs/images/metadata-dataset.png" 
+          alt="Dataset metadata"
+          class="mt-3 mb-3 border border-info rounded">
+
 
-You can find an example of how to use the logging API in this [notebook](https://github.com/kubeflow/metadata/blob/master/sdk/python/demo.ipynb).
 
-## Backend
+## Backend and REST API
 
-The backend uses [ML-Metadata](https://github.com/google/ml-metadata/blob/master/g3doc/get_started.md) to manage all the metadata and relations. It exposes a [REST API](https://github.com/kubeflow/metadata/blob/master/api/service.swagger.json).
+The Kubeflow metadata backend uses [ML Metadata
+(MLMD)](https://github.com/google/ml-metadata/blob/master/g3doc/get_started.md) 
+to manage the metadata and relationships. 
+
+The backend exposes a 
+[REST API](https://github.com/kubeflow/metadata/blob/master/api/service.swagger.json).
+
+You can add your own metadata types so that you can log metadata for custom
+artifacts. To add a custom type:
+
+* Define your custom metadata type using the same 
+  [schema format](https://github.com/kubeflow/metadata/tree/master/schema) as 
+  the predefined types.
+* Load your custom type by sending a REST API request to the schema registration
+  endpoint. For example, the following request sets up a `data_set` type:
 
-## UI
+    ```
+    curl -X POST http://localhost:8080/v1/type/namespaces/my-company.com/kinds/data_set/apiversions/v1 --header "Content-Type: application/json" -d @data_set_schema.json
```

Comment:
@zhenghuiwang Please would you give me an up to date code snippet for this request.

Reply:
I can send you query as an example soon.

Reply:
The following CURL registers an artifact type of _name_ `myorg/mytype/v1` with three
_properties_: `f1` of string type, `f2` of integer type and `f3` of double type.

```
curl -X POST http://localhost:8080/api/v1alpha1/artifact_types \
  --header "Content-Type: application/json" -d \
  '{"name":"myorg/mytype/v1","properties":{"f1":"STRING", "f2":"INT", "f3": "DOUBLE"}}'
```

if you want to link the endpoint spec, it is
`/docs/reference/metadata/v1alpha1/kubeflow-metadata-api-spec/#operation--api-v1alpha1-artifact_types-post`

Reply:
Thanks!